### PR TITLE
PROD-3261 add support for modals -> master

### PR DESCRIPTION
**Pull request by vas3a**
[PROD-3261](https://topcoder.atlassian.net/browse/PROD-3261)

Adds support for modals within the library, and adds the support modal to show up when you click the support link in the footer.

NOTE: this adds a rather important change. Not sure if we should call it "breaking" and we shouldn't probably move to `v2` either. 
The AuthUser now needs firstName, lastName and email fields. Also, until now I requested "initials" but now I'm composing them based on first&last name. These fields are used in the support modal to prefill the user fields.

The api (readme updated as well) now supports one additional property: `supportMeta` to allow to send additional data for the support ticket (challengeId and isSelfService).

Desktop:
image

mobile:
image

### CODE REVIEW

File: src/lib/components/modals/SupportModal.svelte [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  class={styles.supportModal}
+  bind:isVisible={isVisible}
+  title="WE'RE HERE TO HELP"
+  size="sm"
+>
+  <p>
+    Hi <strong>{user.firstName}</strong>, we're here to help.
+  </p>
+  <p>
+    Please describe what you 'd like to discuss, and a Topcoder Solutions Expert
```
Comment (testflyjets) [inline src/lib/components/modals/SupportModal.svelte:65]: Typo: extra space in "you 'd"

**Review by testflyjets — CHANGES_REQUESTED** [review]
Correct minor typo

**Comment by brooketopcoder** [comment]
@vas3a We'll need to support the logged out state, so these new params must be optional, which means they'd be backwards compatible.

We can communicate the new feature and let Oahn and copilots coordinate when they want to implement them.

**Comment by vas3a** [comment]
@brooketopcoder right! It's mostly like that right now, except the "initials" have been now replaced by first and last names for the authenticated user. The only thing I need to fix right now for the anonymous user is to not show "undefined" in the form (which is easy fix).

That's why I called it an important change, but not really breaking change.

**Review by testflyjets — APPROVED** [review]

**Review by brooketopcoder — APPROVED** [review]

**Pull request merged**